pkg/catalog/manager/catalog_preprocess.go

```diff
@@ -0,0 +1,92 @@
+package manager
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+
+	"github.com/rancher/rancher/pkg/catalog/helm"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+)
+
+const invalidURLErrMsg string = "a valid url must consist of an http, https, or git scheme"
+
+type ChartsWithErrors struct {
+	Template *helm.ChartVersion
+	error    InvalidChartError
+}
+
+type InvalidChartError struct {
+	Err          string
+	TemplateName string
+}
+
+func (e *InvalidChartError) Error() string {
+	return fmt.Sprintf("failed %v %s", e.TemplateName, e.Err)
+}
+
+// Go through each catalog entry and check for errors in values used for template name and labels
+// due to restrictions for kubernetes objects. Do not create templates with errored charts.
+func (m *Manager) preprocessCatalog(catalog *v3.Catalog, catalogEntries map[string]helm.ChartVersions) ([]ChartsWithErrors, map[string]helm.ChartVersions) {
+	var invalidEntries []ChartsWithErrors
+	for chart, metadata := range catalogEntries {
+		for _, version := range metadata {
+			var versionErrors []string
+			// get a chart name that matches helm general conventions
+			templateName := getValidTemplateName(catalog.Name, chart)
+			// check that name could be used as a kubernetes qualified name
+			errorString := validation.IsQualifiedName(templateName)
+			if len(errorString) > 0 {
+				versionErrors = append(versionErrors, fmt.Sprintf("template name %s: [%s]", templateName, strings.Join(errorString, ";")))
+			}
+			// check that name can be used as a label
+			errorString = validation.IsValidLabelValue(templateName)
+			if len(errorString) > 0 {
+				versionErrors = append(versionErrors, fmt.Sprintf("template name %s: [%s]", templateName, strings.Join(errorString, ";")))
+			}
+			// validate label that follows general conventions and can be used as kubernetes label
+			templateVersionName := getValidTemplateNameWithVersion(templateName, version.Version)
+			errorString = validation.IsDNS1123Subdomain(templateVersionName)
```

Comment:
Is this the correct check here? Should it also be IsQualifiedName? And the comment above this needs to be updated

Reply:
both checks need to be the same and they are not fully qualified